collections/src/androidTest/java/solid/collections/SolidMapTest.java

```diff
@@ -180,7 +180,7 @@ public void testToString() throws Exception {
     }
 
     private <T> void assertSetEquals(Set<T> expected, Set<T> actual) {
-        assertArrayEquals(expected.toArray(), actual.toArray());
+        assertEquals(expected, actual);
```

Comment:
This won't result in a nice failure report as `assertArrayEquals` would have, but at least won't create false negatives. I'd suggest using `assertJ`, they've got assertions for anything you wish, way richer than default junit assertions. Their API is more fluent as well.